src/hazelcore/Statics_Pat.rei

```diff
@@ -16,6 +21,7 @@ let ana_nth_type_mode:
  * produces a new context with bindings introduced by `p` (if possible)
  */
 let syn: (Contexts.t, UHPat.t) => option((HTyp.t, Contexts.t));
+let syn_moded: (Contexts.t, UHPat.t) => option(HTyp.t);
```

Comment:
naming is a bit confusing here -- was expecting this to take a pattern_var_mode input, but "moded" here refers to the ModedVariable constructor of that type, and it isn't obvious that to get UnknownVariable you just use the default. Thinking it might be better for external clients to have to be explicit about which choice they are making when they call top-level syn...